### ISSUE TITLE
Alexa and Haaska Issue

### ISSUE THREAD

**Issue by intrawit** [auchter/haaska#32]
i followed exactly what bruhautomation said but i get the error message when i run the test.

{
  "errorMessage": "Unable to import module 'lambda_function'"
}

i did upload the haaska file from my pi
clicked save and test and then i get the errormessage.

please help

**Comment by intrawit** [comment]
what must the handler be in aws?

**Comment by gabrielbratescu** [comment]
The handler should be:  haaska.event_handler

**Comment by intrawit** [comment]
What must the redirect urls be on developer account to link Alexa with home
assistant?

**Comment by gabrielbratescu** [comment]
You find the redirect URLs in the Configuration page of the skill.
Something like https://layla.amazon.com/api/..... and https://pitangui.amazon.com/api.....

**Comment by intrawit** [comment]
But those redirect urls is not working.. That is why I am asking.

**Comment by gabrielbratescu** [comment]
In what sense are they not working ? You have to copy them to Amazon/your app/web settings/Allowed Return URL

**Comment by intrawit** [comment]
I copy them from the developer skill page to the login to Amazon account.
I even tried it to put it in the Web browser to test the URL but it just
shows me a white page and says message= null

**Comment by gabrielbratescu** [comment]
It's on to return null, because in browser you are not sending the required information.
Go to next step. Please be aware that the video from bruhautomation has some missing info, so double check the video with setup info from the first page of this project.

**Comment by intrawit** [comment]
Where can I get the full info for the full project?

**Comment by gabrielbratescu** [comment]
I don not know what you mean by that.
Today I was able to set it up by using bruh's video and the setup info from this project.
It's not a walk in the park, but it works.
If you need some help, send me a private message and maybe we can do somenthing.

**Comment by mzjlswatt** [comment]
@gabrielbratescu
Hi,did u successfully use the Alex?i fellowed the video step by step,nothing missed I think,finally my Alex app need account linking,I don't know what's the account!

**Comment by intrawit** [comment]
The Alexa is working on basic commands.it just when I ask Alexa to link
through to the home assistant to switch of lights..that doesn't work.

**Comment by gabrielbratescu** [comment]
@mzjlswatt Yes, it works. I can say Alexa turn on /off "device from HA".  Except a few failures, see issue #34 . 
I don't know for sure what do you mean by account linking. When you start Alexa app you need to login with your amazon account. And then install the skill you just created (haaska). You find it under Skills/Your skills.  After that it tells you you need to discover your devices, do so. A list of discovered devices will the appear in Alexa.
Here I had some issues because the I had some devices without friendly_name or haaska_name and Haaska would generate an error on discovery. The way the script is working right now is that an error occures on one device, it stops and no devices are discovered. All or nothing. Please check pull request #29 . I made this modification to my script in order to work.

@intrawit After installing Haaska skill, did Alexa discovered your devices ? Do you see them in Alexa/Smart Home menu ? Please read also the above response.  

For the future, please tell me the minute from the bruh's video where are are stuck on. I am not a native english and it's easier for me to understand at what step are you

**Comment by iyesildirek** [comment]
I am not able to enable the haaska skill on my amazon app. I run make test on my RPI and I get the following results:
> Unknown output type: ENTER
> {
>   "header": {
>     "payloadVersion": "2",
>     "namespace": "Alexa.ConnectedHome.System",
>     "name": "HealthCheckResponse",
>     "messageId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
>   },
>   "payload": {
>     "isHealthy": true
>   }
> }
> true

When I got to my AWS console and test my skill I get the following in my log output:
> Execution result: succeeded(logs)
> 
> START RequestId: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx Version: $LATEST
> END RequestId: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> REPORT RequestId: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx	Duration: 464.19 ms	Billed Duration: 500 ms

I can also see a list of all my devices in my AWS Console log output:
> The area below shows the result returned by your function execution.
> {
>   "header": {
>     "payloadVersion": "2",
>     "namespace": "Alexa.ConnectedHome.Discovery",
>     "name": "DiscoverAppliancesResponse",
>     "messageId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxx"
>   },
>   "payload": {
>     "discoveredAppliances": [
>       {
>         "modelName": "Unknown",
>         "actions": [
>           "turnOn",
>           "turnOff",
>           "incrementPercentage",
>           "decrementPercentage",
>           "setPercentage"
>         ],
>         "friendlyDescription": "Home Assistant Light",
>         "version": "Unknown",
>         "additionalApplianceDetails": {
>           "entity_id": "light.hue_white_lamp_1",
>           "dimmable": true
>         },
>         "manufacturerName": "Unknown",
>         "friendlyName": "Hue white lamp 1",
>         "applianceId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
>         "isReachable": true
>       },

I have been stuck trying to figure out what is wrong with my configuration and why am I getting an "account linkage required" message below the button to enable my Skill.

Any ideas what I am doing wrong?

**Comment by trisk** [comment]
@iyesildirek "Account linkage required" is normal for this skill, which requires OAuth (Login with Amazon). When you click "Enable", you will be sent to the Login with Amazon page for the security profile you created.

**Comment by mohaggag** [comment]
Do i need a real echo to run this skill . I mean the discovery command is only available on a real amazon echo but i am using The AVS on a raspberry pi so I don't know if this skill will work or not

**Comment by trisk** [comment]
@mohaggag I believe it will work if you can access the Smart Home web interface for Alexa while signed into your Amazon account, http://alexa.amazon.com/spa/index.html#smart-home

**Comment by mohaggag** [comment]
@trisk no not working

**Comment by trisk** [comment]
@mohaggag can you be more specific? Does the Smart Home page list the skill you created and the "Discover Devices? link?

**Comment by mohaggag** [comment]
@trisk yes

**Comment by redstar2k1** [comment]
@mohaggag so what is "not working"? No devices shows up in the "Smart Home" page after you try to "Discover Devices"? If so, what country/region is your Amazon Developer account set (https://developer.amazon.com/settings/profile/detail.html) and on what region (North America/Europe) is your Lambda skill set?

**Comment by shadow05139** [comment]
replace whatever is in the handler box with: 

haaska.event_handler
